pkg/wallet/wallet.go

```diff
@@ -0,0 +1,81 @@
+package wallet
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+
+	"github.com/massalabs/thyra/pkg/node/base58"
+	"lukechampine.com/blake3"
+)
+
+const VERSION byte = 0x0
+
+type Wallet struct {
+	PrivateKey ed25519.PrivateKey
+	Address    string
+}
+
+func New() (*Wallet, error) {
+
+	pubKey, privKey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		return nil, errors.New("unable to create a new wallet")
+	}
+	//generate the wallet address from public key
+	address := getAddressFromPubKey(pubKey)
+	w := &Wallet{
+		PrivateKey: privKey,
+		Address:    address,
+	}
+	return w, nil
+}
+
+//NewFromSeed expects base58 encoded secreat key as per massa and returns the wallet
+func NewFromSeed(key string) (*Wallet, error) {
```

Comment:
I'm not sure to understand what is the purpose of this function other than reproductibility. If it's only for that, I don't understand its usage.